packages/simulators/lib/simulator_manager.dart

```diff
@@ -0,0 +1,214 @@
+
+    // The simulator list, have the version string followed by a list of phone
+    // names along with their ids and their statuses. Example output:
+    // -- iOS 13.5 --
+    //   iPhone 8 (2A437C91-3B85-4D7B-BB91-32561DA07B85) (Shutdown)
+    //   iPhone 8 Plus (170207A8-7631-4CBE-940E-86A7815AEB2B) (Shutdown)
+    //   iPhone 11 (7AEC5FB9-E08A-4F7F-8CA2-1518CE3A3E0D) (Booted)
+    //   iPhone 11 Pro (D8074C8B-35A5-4DA5-9AB2-4CE738A5E5FC) (Shutdown)
+    //   iPhone 11 Pro Max (3F33AD9A-805E-43E0-A86C-8FC70464A390) (Shutdown)
+    // -- iOS 13.6 --
+    //   iPhone 8 (2A437C91-3B85-4D7B-BB91-32561DA07B85) (Shutdown)
+    //   iPhone 8 Plus (170207A8-7631-4CBE-940E-86A7815AEB2B) (Shutdown)
+    //   iPhone 11 (7AEC5FB9-E08A-4F7F-8CA2-1518CE3A3E0D) (Booted)
+    //   iPhone 11 Pro (D8074C8B-35A5-4DA5-9AB2-4CE738A5E5FC) (Shutdown)
+    final int indexOfVersionListStart =
+        simulatorsList.indexOf(simulatorVersion);
+    final String restOfTheOutput = simulatorsList
+        .substring(indexOfVersionListStart + simulatorVersion.length);
+    final int indexOfNextVersion = restOfTheOutput.indexOf('--');
+    final String listOfPhones =
+        restOfTheOutput.substring(0, indexOfNextVersion);
+
+    final int indexOfPhone = listOfPhones.indexOf(phone);
+    if (indexOfPhone == -1) {
+      throw Exception('Simulator of $phone is not available for iOS version '
+          '${osMajorVersion}.${osMinorVersion}');
+    }
+
+    final String phoneInfo = listOfPhones.substring(indexOfPhone);
+    final int endIndexOfPhoneId = phoneInfo.indexOf(')');
+    final String simulatorId =
+        phoneInfo.substring(phoneInfo.indexOf('(') + 1, endIndexOfPhoneId);
+
+    final String phoneInfoAfterId = phoneInfo.substring(endIndexOfPhoneId + 1);
+    final String simulatorStatus = phoneInfoAfterId.substring(
+        phoneInfoAfterId.indexOf('(') + 1, phoneInfoAfterId.indexOf(')'));
+    return IosSimulator._(simulatorStatus == 'Booted', simulatorId);
+  }
+
+  Future<String> _listExistingSimulators(
+      int osMajorVersion, int osMinorVersion) async {
+    final io.ProcessResult versionResult =
+        await io.Process.run('xcrun', ['simctl', 'list']);
+
+    if (versionResult.exitCode != 0) {
+      throw Exception('Failed to list iOS simulators.');
+    }
+    final String output = versionResult.stdout as String;
+    // If the requested iOS version simulators exists, there should be a block
+    // starting with: `-- iOS osMajorVersion.osMinorVersion --`
+    final bool versionCheck =
+        output.contains('-- iOS ${osMajorVersion}.${osMinorVersion} --');
+
+    if (!versionCheck) {
+      throw Exception(
+          'Requested simulator version iOS ${osMajorVersion}.${osMinorVersion} '
+          'is not available.');
+    }
+
+    return output;
+  }
+}
+
+/// A class that can be used to boot/shutdown an iOS Simulator.
+class IosSimulator {
+  final String _id;
+  String get id => _id;
```

Comment:
nit: if the field is final you don't need to split into field + getter. `final String id` has the exact same effect.

Reply:
thanks! done